direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -356,16 +381,24 @@ private void submitConsumerWithObserver(
             consumerRef.set(kafka);
 
             // we need to poll first to initialize kafka assignments and rebalance listener
-            ConsumerRecords<Object, Object> poll = kafka.poll(pollDuration);
+            ConsumerRecords<Object, Object> poll;
+            Map<TopicPartition, Long> endOffsets;
 
-            Map<TopicPartition, Long> endOffsets =
-                stopAtCurrent ? findNonEmptyEndOffsets(kafka) : null;
+            do {
+              poll = kafka.poll(pollDuration);
+              endOffsets = stopAtCurrent ? findNonEmptyEndOffsets(kafka) : null;
 
-            if (log.isDebugEnabled()) {
-              log.debug("End offsets of current assignment {}: {}", kafka.assignment(), endOffsets);
-            }
+              if (log.isDebugEnabled()) {
+                log.debug(
+                    "End offsets of current assignment {}: {}", kafka.assignment(), endOffsets);
+              }
 
-            listener.onPartitionsAssigned(kafka.assignment());
+              listener.onPartitionsAssigned(kafka.assignment());
```

Comment:
Why is needed this cycle?

Reply:
Assignment of partitions for subscriptions with regex apparently takes multiple calls to poll().